src/main/java/redis/clients/jedis/DefaultJedisClientConfig.java

```diff
@@ -113,6 +113,25 @@ public HostAndPortMapper getHostAndPortMapper() {
     return hostAndPortMapper;
   }
 
+  /**
+   * Check if the client name characters are legal
+   * @param name the name
+   * @return Returns true if legal, false otherwise
+   */
+  public static boolean validateClientName(String name) {
```

Comment:
This validation should be done in Connection. Otherwise, any other implementation of JedisClientConfig will not go through this validation.